cunumeric/module.py

```diff
@@ -3446,6 +3446,38 @@ def diagonal(
     )
 
 
+@add_boilerplate("a", "indices", "values")
+def put(
+    a: ndarray, indices: ndarray, values: ndarray, mode: str = "raise"
+) -> None:
+    """
+    Set storage-indexed locations to corresponding values.
+
+    Parameters
+    ----------
+    a : array_like
+        Array to put data into
+    indices : array_like
+        Target indices, interpreted as integers.
+    values : array_like
+        Values to place in `a` at target indices.
```

Comment:
The original documentation also states that "If v is shorter than ind it will be repeated as necessary." We should mention this here as well (assuming we are keeping this behavior).

Reply:
Thank you, I will add this. Yes. we are keeping this behavior